app/src/main/java/com/duckduckgo/app/browser/httpauth/WebViewHttpAuthStore.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.httpauth
+
+import android.os.Build
+import android.webkit.WebView
+import android.webkit.WebViewDatabase
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import com.duckduckgo.app.browser.httpauth.db.HttpAuthDao
+import com.duckduckgo.app.browser.httpauth.db.HttpAuthEntity
+import com.duckduckgo.app.fire.fireproofwebsite.data.FireproofWebsiteDao
+import com.duckduckgo.app.fire.fireproofwebsite.data.website
+import com.duckduckgo.app.global.DispatcherProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+
+data class WebViewHttpAuthCredentials(val username: String, val password: String)
+
+@VisibleForTesting
+fun WebView.setHttpAuthUsernamePasswordCompat(host: String, realm: String, username: String, password: String) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
This is the same logic we have in `DatabaseModule`. I wonder if having it duplicated here without a clear connection between them can lead to possible bugs if something changes. is there any way to unify them or to connect them somehow? 

Checking this twice, it's like we have 2 approaches for the same. 
a) compat method that handles Android api version
b) we call this method after checking that `httpAuthDao` was null, and `httpAuthDao` is `null` based on the device Android Api version.

Reply:
This was on purpose. 
The webview extensions are disconnected from the DAO and viceversa. So we should not remove API checking from one of them assuming it will only be called from the other one and that other one will perform the checking.
If we remove the API checking from one of them and rely on the other one, we'd be coupling this two classes, something that I think is generally undesirable.